### DISCUSSION TITLE
Social cards custom layouts do not show on social media

### DISCUSSION THREAD

**Discussion by buckwem** [squidfunk/mkdocs-material#6947] [Q&A] [answered]
I'm trying to debug an issue where the social tiles are created using Insiders into .cache on my laptop (Windows 11) using mkdocs serve. When I do the same using a GitHub Actions script, no social tiles are generated into assets/images but if I go back to using the public version in GitHub, the basic social tiles are generated, I have validated the differences and made sure Python and every Python library are exactly the same version.

What else should I be checking? Any other ideas for what would allow Insiders custom social tiles to work on mkdocs serve locally but not mkdocs gh-deploy --force?

**Comment by buckwem** [comment]
A few things so far:

1. pngquant needs installing - the note on the Insider [GitHub Actions script](https://squidfunk.github.io/mkdocs-material/publishing-your-site/?h=github+ac#with-github-actions) needs this to be added
3. "apt-get install pngquant"  needs to be "sudo apt-get install pngquant" in the same script
4. I had some images with "\\" when they should be "/". Errors were shown from running gh-deploy but there was no error from running the GitHub Actions workflow (so I did not see it).

**Comment by buckwem** [comment]
All the social tile assets are generated and now in the right place for [securityarchitecture.cloud](https://securityarchitecture.cloud) but no tiles are displayed and when using the [Ahrefs SEO toolbar](https://ahrefs.com/) it doesn't find the open graph tags. So far I can't see anything in the verbose logs. Will have another look later.

**Comment by buckwem** [comment]
The page has not got any Open Graph meta property tags. So what would create the social images but not the tags?

**Comment by squidfunk** [comment]
Thanks for reporting. It is very likely an issue with configuration. Without knowing how you configured your project, the problem could be hiding anywhere. Are you using [custom layouts](https://squidfunk.github.io/mkdocs-material/setup/setting-up-social-cards/#customization)? How did you configure the plugin? Can you please provide a [_minimal_ reproduction](https://squidfunk.github.io/mkdocs-material/guides/creating-a-reproduction/)? Otherwise we can only guess, and my guess is that, looking at your site, the social plugin is not enabled 😅

For errors that you encounter we definitely need minimal reproductions to fix them, which we'd be more than happy to.

**Reply by squidfunk** [reply]
For instance, [this is the configuration](https://github.com/squidfunk/mkdocs-material-insiders/blob/master/mkdocs.yml#L23) we use for our documentation to generate social cards (reduced):

``` yaml
site_url: https://squidfunk.github.io/mkdocs-material/
plugins:
  - social:
      cards_layout: default/variant
```

Make sure `site_url` is set, [as mentioned in our documentation](https://squidfunk.github.io/mkdocs-material/setup/setting-up-social-cards/#built-in-social-plugin) (info box).

**Reply by buckwem** [reply]
Thanks Martin - I thought you might suggest that. Yes, I have made both those changes.

I created [https://github.com/buckwem/social-test](https://github.com/buckwem/social-test)
that is [here](https://buckwem.github.io/social-test/).

If I swap between insiders and the public version in ci.yml, I can see meta properties are created for the public code but not insiders.

Just one thing that occurred to me - would using .pages files affect building the meta properties?

If you want me to further cut it down, I'm happy to do that.

**Reply by squidfunk** [reply, marked as answer]
Thanks. If I'm not missing something, my question about [custom layouts](https://squidfunk.github.io/mkdocs-material/setup/setting-up-social-cards/#customization) in https://github.com/squidfunk/mkdocs-material/discussions/6947#discussioncomment-8871843 was not answered, and this is the problem here. The custom layout is missing [the `tags` setting](https://squidfunk.github.io/mkdocs-material/setup/setting-up-social-cards/#tags) as mentioned in our documentation. If you add them, you should see them appear and the social networks correctly pick up the social cards. We'll try to make it clearer once we rewrite the documentation on social cards that this is mandatory. We decided to let the layout define what tags should be added to give maximum flexibility, but I understand that this might be easy to miss.

**Reply by buckwem** [reply]
Thanks for taking the time out to respond on a Saturday. That worked!

So my understanding is for the public release, tags are turned on for just for the social cards when they are used. However, using custom social cards, you will need to explicity enable the tags plugin and add the definition of tags to your layout file.

Would it help if I propose a PR to update the documentation? I can summarise some proposed changes as here are a few other areas I think could be helped with some additional information. If you agree, then I can write the actual improvements.

**Reply by squidfunk** [reply]
Awesome!

> So my understanding is for the public release, tags are turned on for just for the social cards when they are used. However, using custom social cards, you will need to explicity enable the tags plugin and add the definition of tags to your layout file.

Custom layouts always mandate that you define which tags you want to use. The community edition does not include the custom layouts functionality, as the social plugin has been rewritten from the ground up.

> Would it help if I propose a PR to update the documentation? I can summarise some proposed changes as here are a few other areas I think could be helped with some additional information. If you agree, then I can write the actual improvements.

Yes, you can create a PR, but maybe we can discuss first what you want to update, so that we can synchronize efforts ☺️ I guess it'll build on https://github.com/squidfunk/mkdocs-material/discussions/6947#discussioncomment-8870235, but maybe you can elaborate a bit more.
